backend/src/shortener/shortener.service.ts

```diff
@@ -33,30 +50,98 @@ export class ShortenerService {
    * Generating the short url
    * @returns {String} Returns a random 5 characters url
    */
-  generateShortUrl = (): string => {
+  generateShortenedUrl = (): string => {
     return Math.random().toString(36).substring(2, 7);
   };
 
   /**
-   * Check if the short url is not in use.
-   * @param {String} shortUrl The shorten url.
-   * @returns {Boolean} Is the shortener url taken
+   * Checks if a shortened URL is available (not in use).
+   * @param {string} shortenedUrl - The shortened URL.
+   * @returns {Promise<boolean>} - True if the shortened URL is available, false otherwise.
    */
-  isShortUrlAvailable = async (shortUrl: string): Promise<boolean> => {
-    const originalUrl = await this.getOriginalUrl(shortUrl);
+  isShortenedUrlAvailable = async (shortenedUrl: string): Promise<boolean> => {
+    const originalUrl = await this.getOriginalUrl(shortenedUrl);
     return originalUrl === null;
   };
 
   /**
    * Add the short url to the server routes.
    * @param {String} originalUrl The original url.
-   * @param {String} shortUrl The shorten url.
+   * @param {String} shortenedUrl The shorten url.
+   */
+  addUrl = async (originalUrl: string, shortenedUrl: string) => {
+    const isShortenedUrlAvailable = await this.isShortenedUrlAvailable(shortenedUrl);
+    if (!isShortenedUrlAvailable) {
+      throw new Error('ShortenedURL already taken');
```

Comment:
Change the error message to be easier to read - 
Shortened Url is already taken

Reply:
done